### PR TITLE
Fix/2139 jest snapshots

### LINKED ISSUE #2139: @emotion/jest - Snapshot serializer is not replacing hashes
**Current behavior:**

I have updated project to emotion 11 and i have noticed failing jest tests. When i provide emotion styles as an array: `css={[style1, style2]}`, their hashes in snapshot will not be replaced.

Component with emotion styles: https://github.com/ejdzipi/emotion-jest-serializer-repro/blob/master/src/StyleArrayComponent/StyleArrayComponent.jsx

Resulting snapshot: https://github.com/ejdzipi/emotion-jest-serializer-repro/blob/master/src/StyleArrayComponent/__snapshots__/StyleArrayComponent.test.js.snap
 
**To reproduce:**

I have prepared a repro: https://github.com/ejdzipi/emotion-jest-serializer-repro

**Expected behavior:**

Hashes are correctly replaced

**Environment information:**

- `react` version: 16.14
- `@emotion/react` version: 11.1.1

### PR DESCRIPTION
**Pull request by ajs139**
**What**:

Fix #2139 - Snapshot serializer is not replacing hashes.

**Why**:

When creating snapshots for components that have an array as the css property, the snapshot was outputting some of the  internal properties. Additionally, there was a problem with the serialization when using `mount` on components that had the css properties on child components. 

**How**:

Fixed a bug where the component wasn't being unwrapped properly (so its children were not being modified).
Accounted for the `css` prop being an array.

**Checklist**:

- [ ] Documentation N/A
- [x] Tests
- [x] Code complete
- [x] Changeset

### CODE REVIEW

File: packages/jest/test/__snapshots__/react-enzyme.test.js.snap [hunk 1]
```diff
@@ -114,6 +114,33 @@ exports[`enzyme mount theming 1`] = `
...
+  background-color: black;
+  color: white;
+}
+
+<div
+  className="emotion-0"
+>
+  Test content
+  <div
+    className="emotion-1 emotion-2"
```
Comment (ajs139) [inline packages/jest/test/__snapshots__/react-enzyme.test.js.snap:135]: need to look at why this extra class name is present

File: packages/jest/test/__snapshots__/react-enzyme.test.js.snap [hunk 2]
```diff
@@ -343,6 +368,24 @@ exports[`enzyme shallow theming 1`] = `
...
+  display: -ms-flexbox;
+  display: flex;
+}
+
+<div
+  className="emotion-0"
+>
+  Test content
+  <div
+    className="emotion-1 emotion-2"
```
Comment (ajs139) [inline packages/jest/test/__snapshots__/react-enzyme.test.js.snap:384]: will look at this also

File: yarn.lock [hunk 3]
```diff
@@ -1866,6 +1866,11 @@
   resolved "https://registry.yarnpkg.com/@emotion/memoize/-/memoize-0.7.3.tgz#5b6b1c11d6a6dddf1f2fc996f74cf3b219644d78"
   integrity sha512-2Md9mH6mvo+ygq1trTeVp2uzAKwE2P7In0cRpD/M9Q70aH8L+rxMLbb3JCN2JoSWsV2O+DdFjfbbXoMoLBczow==
 
+"@emotion/memoize@0.7.4":
```
Comment (ajs139) [inline yarn.lock:1869]: will remove this

File: packages/jest/test/__snapshots__/react-enzyme.test.js.snap (hunk 1 above)
Comment (Andarist) [reply packages/jest/test/__snapshots__/react-enzyme.test.js.snap]: It seems that you have already resolved this issue, right?

File: packages/jest/test/react-enzyme.test.js [hunk 4]
```diff
@@ -191,6 +204,25 @@ describe('enzyme', () => {
     cases
   )
 
+  test('child containing css array property', () => {
```
Comment (Andarist) [inline packages/jest/test/react-enzyme.test.js:207]: is this the correct test title? no css prop here holds an array

File: packages/jest/test/react-enzyme.test.js [hunk 5]
```diff
@@ -143,6 +143,19 @@ const cases = {
       )
     }
   },
+  'using the css helper': {
```
Comment (Andarist) [inline packages/jest/test/react-enzyme.test.js:146]: OTOH this one holds an array so I would assume that maybe the added test titles should be swapped but I'm also not sure what "using the css helper" test would be about - could you make this test title a little bit more descriptive?

File: packages/jest/test/react-enzyme.test.js (hunk 4 above)
Comment (ajs139) [reply packages/jest/test/react-enzyme.test.js]: updated, as noted the titles got switched

File: packages/jest/test/react-enzyme.test.js (hunk 5 above)
Comment (ajs139) [reply packages/jest/test/react-enzyme.test.js]: thanks, let me know if the new description is missing anything

File: .changeset/two-laws-provide.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'@emotion/jest': patch
+---
+
+Fix issue creating enzyme snapshots for components using an array as the css prop.
```
Comment (Andarist) [inline .changeset/two-laws-provide.md:5]:
```suggestion
---
'@emotion/jest': minor
---

Fixed an issue with Enzyme snapshots for components using an array as the `css` prop - those should be printed OK now.
```

**Review by Andarist — APPROVED** [review]

**Pull request merged**